config/configqueue/queue.go

```diff
@@ -0,0 +1,69 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package configqueue // import "go.opentelemetry.io/collector/config/configqueue"
+
+import (
+	"errors"
+
+	"go.opentelemetry.io/collector/component"
+)
+
+// QueueConfig defines configuration for queueing requests before exporting.
+// It's supposed to be used with the new exporter helpers New[Traces|Metrics|Logs]RequestExporter.
+// This API is at the early stage of development and may change without backward compatibility
+// until https://github.com/open-telemetry/opentelemetry-collector/issues/8122 is resolved.
+type QueueConfig struct {
```

Comment:
second thoughts: for retry we put it under config because receivers/connectors may also retry adding to a pipeline. For queue do we see an use case to be used outside exporter? If not, consider to have a `exporter/exporterqueue` where we declare the config and the Queue interface?

Reply:
Sounds good to me. I don't think there is any use case outside exporters

Reply:
I moved the new API to `exporter/exporterqueue`. Given that both `exporter/exporterhelper` and `exporter/exporterqueue` depend on `exporter/exporterqueue/internal`, I moved it as is to `exporter/internal` for now to keep the changeset simple. Once this PR is merged, I'll distribute that code.